pyccel/codegen/printing/fcode.py

```diff
@@ -612,25 +612,173 @@ def _print_PythonImag(self, expr):
     #========================== Numpy Elements ===============================#
 
     def _print_NumpySum(self, expr):
-        return expr.fprint(self._print)
+        """Fortran print."""
+
+        rhs_code = self._print(expr.arg)
+        return 'sum({0})'.format(rhs_code)
 
     def _print_NumpyProduct(self, expr):
-        return expr.fprint(self._print)
+        """Fortran print."""
+
+        rhs_code = self._print(expr.arg)
+        return 'product({0})'.format(rhs_code)
 
     def _print_NumpyMatmul(self, expr):
-        return expr.fprint(self._print)
+        """Fortran print."""
+        a_code = self._print(expr.a)
+        b_code = self._print(expr.b)
+
+        if expr.a.order and expr.b.order:
+            if expr.a.order != expr.b.order:
+                raise NotImplementedError("Mixed order matmul not supported.")
+
+        # Fortran ordering
+        if expr.a.order == 'F':
+            return 'matmul({0},{1})'.format(a_code, b_code)
+
+        # C ordering
+        return 'matmul({1},{0})'.format(a_code, b_code)
+
+    def _print_NumpyImag(self, expr):
+        """Fortran print."""
+
+        value = self._print(expr.arg)
+        code = 'aimag({0})'.format(value)
+        return code
```

Comment:
This class inherits from `PythonImag` now. Does it need a specialised `_print_NumpyImag` method or does the `_print_PythonImag` function do the same thing?

Reply:
yeah it does the same as `_print_PythonImag`, so can we remove it?.

Reply:
When looking for the function to print a `NumpyImag` the `FCodePrinter` will call `_print_NumpyImag` if it exists, and if it doesn't then it will call `_print_PythonImag`

Reply:
I think i will keep the code readable and make `_print_NumpyImag` call `_print_PythonImag`, what do you think ?.

Reply:
I think this is a bad idea. It goes against the pattern established elsewhere. It leads to redundant code, and if we used your method and wanted to rename a class, this would add a lot more changes. For every other class we only implement the functions that are absolutely necessary in order to reduce the amount of code